### PR TITLE
[WIP][Marketplace] - Marketplace Shop switch dropdown, seller homepage, marketplace optimisation 

### PR DESCRIPTION
**Pull request by lcampanis**
Introducing a shop switch dropdown to allow the owner, in a marketplace setup, to switch between shops in the homepage grid.

Shops now have their own url at `//shop/<shopId>` 

The following scenarios are in place

**Normal Grid (all sellers)**
image

**My Shop (owner shop/products)**
image

**All other sellers**
image

image

### CODE REVIEW

**Comment by lcampanis** [comment]
@aaronjudd We will be fixing conflicts and pushing a whole lot of fixes later on today for seller permissions all throughout, including PDP, createShop, validations, security and all media issues. Practically all seller permissions should be in place against other guests, anonymous, owners and admins.

**Comment by mikemurray** [comment]
If you could resolve the conflicts I will check this out tonight / tomorrow

**Comment by lcampanis** [comment]
@mikemurray apologies for the delay, just swamped with the soft launch now. We aim to have all optimisation and permission fixes in here by Monday, which solve a number of issues with sellers, media, etc. We'll then bring all branches to date as well. Also preparing a new WIP PR for stripe managed accounts on the side, which optimises and extends the existing stripe connect PR with new managed accounts. ;)

**Comment by spencern** [comment]
Hey @lcampanis @prosf 
Are there any additional updates coming for this PR that I should be aware of before merging? Looking to get this PR merged into `marketplace` soon if everything is ready to go so we can continue to make headway on the marketplace side of RC.

File: imports/plugins/included/marketplace/client/templates/settings/sellerShopSettings.js [hunk 1]
```diff
@@ -83,47 +80,39 @@ Template.sellerShopSettings.helpers({
...
+    if (!sellerShop) {
+      return;
+    }
 
-      if (!sellerShop) {
-        return;
-      }
+    const currencies = [];
+    for (const currency in sellerShop.currencies) {
+      if ({}.hasOwnProperty.call(sellerShop.currencies, currency)) {
```
Comment (spencern) [inline imports/plugins/included/marketplace/client/templates/settings/sellerShopSettings.js]: This pattern is clever and brief, but may be less understandable than using something like `Object.prototype.hasOwnProperty.call`

Comment (lcampanis) [inline imports/plugins/included/marketplace/client/templates/settings/sellerShopSettings.js]: We thought this was used enough times in RC to carry on with it. Happy to change it, but maybe we can setup a standard for it and change them all?
image

**Comment by lcampanis** [comment]
@spencern nothing else vital for this branch. Please feel free to merge and close. 
We'll do some small PRs next week on bug fixing and a big one on Stripe Connect Managed Accounts, which will be wip.
The checkout flow needs to be analysed by RC based on the direction taken. We have a rough idea of what it will be, but please let us know and we'll start sharing fixes around that.

File: imports/plugins/included/marketplace/client/templates/shops/shopSelect.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  });
+});
+
+Template.shopSelect.helpers({
+  sellerShops() {
+      const currentShopId = Reaction.Router.getParam("shopId") || 0;
+      const selector = {
+        // ignore blank site
+        _id: {
+          $ne: "ddzuN2YPvgvx7rJS5"
```
Comment (aaronjudd) [inline imports/plugins/included/marketplace/client/templates/shops/shopSelect.js]: Yeah, this feels hokey.. the only point of the seed data having two shops was to force all tests, UI,etc to always be dealing with two shops in the data.  I solution I'd prefer would be be able add a `enabled` property for the shop, to allow a shop to be disabled, and then filtered here.

**Pull request merged**